packages/expo-cli/src/commands/init.ts

```diff
@@ -362,19 +362,43 @@ export async function initGitRepoAsync(
   flags: { silent: boolean; commit: boolean } = { silent: false, commit: true }
 ) {
   // let's see if we're in a git tree
+  let insideGitTree = false;
+
   try {
-    await spawnAsync('git', ['rev-parse', '--is-inside-work-tree'], {
+    const resultPromise = await spawnAsync('git', ['rev-parse', '--is-inside-work-tree'], {
       cwd: root,
     });
-    !flags.silent && Log.log('New project is already inside of a git repo, skipping git init.');
+
+    insideGitTree = resultPromise.stdout.trim() === 'true';
   } catch (e) {
     if (e.errno === 'ENOENT') {
       !flags.silent && Log.warn('Unable to initialize git repo. `git` not in PATH.');
       return false;
     }
   }
 
-  // not in git tree, so let's init
+  let runGitInit = true;
+
+  if (insideGitTree) {
+    const { answer } = await prompts({
+      type: 'text',
+      name: 'answer',
```

Comment:
Ensure this works as expected when running `expo init --non-interactive`

Reply:
After the change to `confirmAsync`, I tested with `expo init --non-interactive` and it works by defaulting to skipping when within a Git tree, and not skipping otherwise.